### PR TITLE
add sample dialog: add all button scb_ex

### LINKED ISSUE #218: NSF Exercise #1 - Formatting the Add Samples window
The 'CANCEL' and 'ADD SAMPLES' buttons need to be moved to the lower right corner. This will mean that they don't conflict with the 'SELECT ALL' button that will go underneath the checkboxes. This should be the case for all exercises within StarCellBio, not just NSF Exercise #1.

### PR DESCRIPTION
**Pull request by annagav**
Fix #218

### CODE REVIEW

File: html_app/scb_ex/addmultipledialog.gss [hunk 1]
```diff
@@ -88,15 +88,26 @@
...
+    position: relative;
+}
+
 
 .scb_ex_button_float {
-    display:block;
-}
\ No newline at end of file
+    position:relative;
+    left:0;
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog.gss]: space after colon

File: html_app/scb_ex/addmultipledialog2.soy [hunk 2]
```diff
@@ -12,72 +12,62 @@
...
-//            {if $assignment.id == 'scb_ex2'}
-
-            <thead class="scb_s_experiment_setup_table_head">
-
-            {foreach $heading in $template.ui.add_multiple_dialog.headings}
-            <td role='columnheader' aria-label='{$heading}' class='scb_s_experiment_setup_table_heading'>
-                {$heading}
-            </td>
-            {/foreach}
+        <table class="scb_s_experiment_setup_table" role='grid' >
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:15]: is there a particular reason for the single quotes for `role='grid'` and all the lines after in the HTML tags?

Comment (annagav) [inline html_app/scb_ex/addmultipledialog2.soy:15]: I don't think so.

File: html_app/scb_ex/addmultipledialog2.soy [hunk 3]
```diff
@@ -12,72 +12,62 @@
...
+            <thead class="scb_s_experiment_setup_table_head" >
+                {foreach $heading in $template.ui.add_multiple_dialog.headings}
+                <td role='columnheader' aria-label='{$heading}'
+                    class="scb_s_experiment_setup_table_heading {if $heading == 'Strain'}scb_s_experiment_setup_table_heading_strain{/if}{if $heading == ''}scb_s_experiment_setup_table_heading_checkbox{/if}" >
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy]: you should reduce the line length using multiple lines

File: html_app/scb_ex/addmultipledialog2.soy [hunk 4]
```diff
@@ -12,72 +12,62 @@
...
-                        {if $cell.kind == 'checkbox'}
-                        <input class='scb_f_experiment_setup_dialog_checkbox' type="checkbox" role='checkbox'  name="{$cell.name}" assignment_id='{$assignment.id}'
-                               experiment_id='{$experiment.id}' treatment_id='{$cell.treatment_id}' cell_line='{$cell_line}' aria-checked='false'>
-                        {/if}
-                    </td>
-                    {/foreach}
-                </tr>
+                        <tr role='row' aria-label='Sample'>
+                            {foreach $cell in $rows.cells}
+                            <td class="scb_s_experiment_setup_table_border {if $cell.kind == 'checkbox'}scb_s_microscopy_usability_checkbox {/if}">
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy]: line length

File: html_app/scb_ex/addmultipledialog2.soy [hunk 5]
```diff
@@ -12,72 +12,62 @@
...
+                    class="scb_s_experiment_setup_table_heading {if $heading == 'Strain'}scb_s_experiment_setup_table_heading_strain{/if}{if $heading == ''}scb_s_experiment_setup_table_heading_checkbox{/if}" >
+                    {$heading}
+                </td>
+                {/foreach}
             </thead>
-
-            <tbody class="scb_s_experiment_setup_table_body">
-                {foreach $cell_line in $template.ui.add_multiple_dialog.order}
+            {foreach $cell_line in $template.ui.add_multiple_dialog.order}
+                {if $cell_line == 'headings' or $cell_line =='order'}
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy]: does this mean that `if $cell_line == 'headings' or $cell_line =='order'` there is no output?
if so you should change it to
`if $cell_line != 'headings' and $cell_line != 'order'` and move the content of the `else` statement here.

**Comment by annagav** [comment]
@giocalitri, I made modifications based on your comments.

File: html_app/scb_ex/addmultipledialog2.soy [hunk 6]
```diff
@@ -12,72 +12,62 @@
...
+                    {if $heading == ''} scb_s_experiment_setup_table_heading_checkbox{/if}" >
+                    {$heading}
+                </td>
+                {/foreach}
             </thead>
-
-            <tbody class="scb_s_experiment_setup_table_body">
-                {foreach $cell_line in $template.ui.add_multiple_dialog.order}
+            {foreach $cell_line in $template.ui.add_multiple_dialog.order}
+                <tbody class="scb_s_experiment_setup_table_body">
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:27]: are you sure this is right? you start `foreach` before `tbody` and you do not close after it.

File: html_app/scb_ex/addmultipledialog2.soy [hunk 7]
```diff
@@ -12,72 +12,62 @@
...
+        <table class="scb_s_experiment_setup_table" role='grid' >
+            <thead class="scb_s_experiment_setup_table_head" >
+                {foreach $heading in $template.ui.add_multiple_dialog.headings}
+                <td role='columnheader' aria-label='{$heading}'
+                    class="scb_s_experiment_setup_table_heading
+                    {if $heading == 'Strain'} scb_s_experiment_setup_table_heading_strain{/if}
+                    {if $heading == ''} scb_s_experiment_setup_table_heading_checkbox{/if}" >
+                    {$heading}
+                </td>
+                {/foreach}
```
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:24]: Did you remove the `if` I commented before?

File: html_app/scb_ex/addmultipledialog2.soy (hunk 6 above)
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:27]: actually, if I counted well, there is an extra `{/foreach}`

Comment (annagav) [inline html_app/scb_ex/addmultipledialog2.soy:27]: I do close `tbody`. It is right under it.

File: html_app/scb_ex/addmultipledialog2.soy (hunk 7 above)
Comment (annagav) [inline html_app/scb_ex/addmultipledialog2.soy:24]: Yes, it is removed.

File: html_app/scb_ex/addmultipledialog2.soy (hunk 6 above)
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:27]: sorry, I missed the extra `{foreach` before the snippet!

File: html_app/scb_ex/addmultipledialog2.soy (hunk 7 above)
Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:24]: Why?

Comment (annagav) [inline html_app/scb_ex/addmultipledialog2.soy:24]: Because I think it is not needed.

Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:24]: I understand that, but can you please explain me why is not needed?

Comment (annagav) [inline html_app/scb_ex/addmultipledialog2.soy:24]: I don't know the history behind that if statement and why it was written in the first place, but I went through the logic and the structure that it is parsing and it seems that such values as 'header' and 'order' will not appear as a value of cell_line.

Comment (giocalitri) [inline html_app/scb_ex/addmultipledialog2.soy:24]: ok, perfect. Merging

**Pull request merged**